### PR TITLE
new: Add `bn` (Bengali) language.

### PR DESCRIPTION
**Pull request by sudipshil9862**

### CODE REVIEW

File: po/bn/messages.po [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+msgid "travel & places"
+msgstr "ভ্রমণ এবং স্থান"
+
+msgctxt "EMOJI GROUP: 6|activities, EMOJI SUB-GROUP: 26|person-activity"
+msgid "activities"
+msgstr "কার্যক্রম"
+
+msgctxt "EMOJI GROUP: 7|objects"
+msgid "objects"
+msgstr "objects"
```
Comment (milesj) [inline po/bn/messages.po:84]: @sudipshil9862  looks like you missed a few translations in this file, or was it intentional?

File: po/bn/messages.po [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+msgid "skin tones"
+msgstr "ত্বক টোন"
+
+msgctxt "EMOJI SUB-GROUP: 32|hair-style"
+msgid "hair styles"
+msgstr "চুলের শৈলী"
+
+msgctxt "EMOJI SUB-GROUP: 33|animal-mammal"
+msgid "mammals"
+msgstr "mammals"
```
Comment (milesj) [inline po/bn/messages.po:224]: another

File: po/bn/messages.po [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+msgid "phone"
+msgstr "ফোন"
+
+msgctxt "EMOJI SUB-GROUP: 70|computer"
+msgid "computer"
+msgstr "কম্পিউটার"
+
+msgctxt "EMOJI SUB-GROUP: 71|light-video"
+msgid "light, film & video"
+msgstr ""
```
Comment (milesj) [inline po/bn/messages.po:376]: this is empty

File: po/bn/messages.po [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+msgid "gender signs"
+msgstr "লিঙ্গ লক্ষণ"
+
+msgctxt "EMOJI SUB-GROUP: 90|math"
+msgid "math symbols"
+msgstr "গণিত প্রতীক"
+
+msgctxt "EMOJI SUB-GROUP: 91|punctuation"
+msgid "punctuation"
+msgstr "punctuation"
```
Comment (milesj) [inline po/bn/messages.po:456]: another

File: packages/data/bn/messages.raw.json [hunk 5]
```diff
@@ -0,0 +1,578 @@
+{
+  "groups": [
+    {
+      "key": "smileys-emotion",
+      "message": "smileys & emotion",
```
Comment (milesj) [inline packages/data/bn/messages.raw.json:5]: these should be the translated messages

**Review by milesj — COMMENTED** [review]
Good start, just a few minor things to fix.

Another thing I forgot to add to the docs, but you'll need to add `.d.ts` files for the data files, as seen here: https://github.com/milesj/emojibase/tree/master/packages/data/ko

File: po/bn/messages.po (hunk 1 above)
Comment (sudipshil9862) [reply po/bn/messages.po:84]: it's a mistake. changing this

File: packages/data/bn/messages.raw.json (hunk 5 above)
Comment (sudipshil9862) [reply packages/data/bn/messages.raw.json:5]: how to create these .d.ts like https://github.com/milesj/emojibase/tree/master/packages/data/ko

Comment (milesj) [reply packages/data/bn/messages.raw.json:5]: just copy them

Comment (sudipshil9862) [reply packages/data/bn/messages.raw.json:5]: does data.json mean data.json.d.ts ?
or should I rename it ?

Comment (milesj) [reply packages/data/bn/messages.raw.json:5]: you can copy the `compact.json.d.ts`, `data.json.d.ts`, and `messages.json.d.ts` files as-is.

`data.json.d.ts` is typescript types for the `data.json` file (which is published).

**Comment by sudipshil9862** [comment]
added .d.ts files:  compact.json.d.ts, data.json.d.ts, messages.json.d.ts
- added some mimssing translation in messages.po

**Comment by sudipshil9862** [comment]
@milesj anyhting else to change ?

**Comment by milesj** [comment]
@sudipshil9862 Yeah just need to fix the failing build and then its good to go

**Comment by sudipshil9862** [comment]
what actually the problem is @milesj ?

```
FAIL  packages/data/tests/shortcodes.test.ts
  ● Test suite failed to run

    Your test suite must contain at least one test.

      at onResult (node_modules/@jest/core/build/TestScheduler.js:[13](https://github.com/milesj/emojibase/actions/runs/6383016280/job/17322854318?pr=161#step:6:14)3:18)
      at node_modules/@jest/core/build/TestScheduler.js:254:19
      at node_modules/emittery/index.js:363:13
          at Array.map (<anonymous>)
      at Emittery.emit (node_modules/emittery/index.js:361:23)
 ```

**Comment by milesj** [comment]
@sudipshil9862 For the shortcode test, we need to update this file: https://github.com/milesj/emojibase/blob/master/packages/regex/shortcode-native.js

And include the regex for bengali.

**Comment by sudipshil9862** [comment]
now where it's failing ?
should I remove `shortcodes` folder inside `/data/bn/` ?

**Comment by milesj** [comment]
@sudipshil9862 Still on the same thing. Looks like bengali may need multiple unicode ranges? Similar to how russian works.

I'm not sure what the range would be though. Just trial and error. I suggest running the tests locally `TEST_LOCALE=bn yarn run test shortcodes`

**Comment by sudipshil9862** [comment]
in gedit, this line is showing : `1FA83": "বুমের‌্যাঙ"`
in vi, this line is showing: `1FA83": "বুমের<200c>যাঙ"`

"‌্" contains a ZERO WIDTH NON-JOINER (U+200C) and a VIRAMA (U+09CD, BENGALI SIGN VIRAMA)

To solve this problem we need a Unicode range `\u2000-\u206F` represents a range of characters and symbols from the "General Punctuation" block within the Unicode Standard.

**Comment by sudipshil9862** [comment]
@milesj please review my PR. all checks have passed

**Comment by milesj** [comment]
@sudipshil9862 Awesome stuff. I'll publish a new version sometime in the next day.

**Pull request merged**